src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -650,7 +650,10 @@ protected override void ProcessRecord()
                     //if fileversion of each process is to be displayed
                     try
                     {
-                        WriteObject(PsUtils.GetMainModule(process).FileVersionInfo, true);
+                        ProcessModule mainModule = PsUtils.GetMainModule(process);
+                        if (mainModule != null) {
```

Comment:
When `mainModule == null`, is the expected behavior to be writing out nothing or throwing an error?
On windows, the current behavior is to throw error when the `FileInfoVersion` is not accessible.

@JamesWTruher Your insight would be helpful here.

Reply:
Related comment from CoreFX https://github.com/dotnet/corefx/blob/f90aa6665a59dc6caaff995a79fb852d21fe6f68/src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs#L176

Reply:
it's also possible on non-windows platforms that mainmodule could be null due to the underlying corefx implementation (ex: process id 0). Throwing an error doesn't seem reasonable.